app/src/main/java/de/tum/in/tumcampusapp/cards/EduroamFixCard.java

```diff
@@ -0,0 +1,148 @@
+package de.tum.in.tumcampusapp.cards;
+
+import android.annotation.SuppressLint;
+import android.annotation.TargetApi;
+import android.app.Notification;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.wifi.WifiConfiguration;
+import android.net.wifi.WifiManager;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.support.v4.app.NotificationCompat;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import java.util.Vector;
+
+import de.tum.in.tumcampusapp.R;
+import de.tum.in.tumcampusapp.activities.SetupEduroamActivity;
+import de.tum.in.tumcampusapp.cards.generic.NotificationAwareCard;
+import de.tum.in.tumcampusapp.managers.CardManager;
+import de.tum.in.tumcampusapp.managers.EduroamManager;
+
+import static de.tum.in.tumcampusapp.managers.EduroamManager.RADIUS_DNS;
+
+
+public class EduroamFixCard extends NotificationAwareCard {
+
+    private Vector<String> errors;
+    private TextView errorsTv;
+    private WifiConfiguration eduroam;
+
+    public EduroamFixCard(Context context) {
+        super(CardManager.CARD_EDUROAM_FIX, context, "card_eduroam_fix_start", false, true);
+    }
+
+    public static CardViewHolder inflateViewHolder(ViewGroup parent) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_eduroam_fix, parent, false);
+        return new CardViewHolder(view);
+    }
+
+    @Override
+    public void updateViewHolder(RecyclerView.ViewHolder viewHolder) {
+        super.updateViewHolder(viewHolder);
+        mCard = viewHolder.itemView;
+        mLinearLayout = (LinearLayout) mCard.findViewById(R.id.card_view);
+        errorsTv = (TextView) mCard.findViewById(R.id.eduroam_errors);
+
+        String txt = "";
+        for (String e : errors) {
+            txt += e + "\n";
+        }
+        errorsTv.setText(txt);
+    }
+
+    @Override
+    protected boolean shouldShow(SharedPreferences prefs) {
+        return !isConfigValid();
+    }
+
+    @Override
+    protected void discard(SharedPreferences.Editor editor) {
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(mContext);
+        prefs.edit().putBoolean("card_eduroam_fix_start", false).apply();
+    }
+
+    @Override
+    protected Notification fillNotification(NotificationCompat.Builder notificationBuilder) {
+        return null;
```

Comment:
I really don't want to bother the user with this. We can always do this at a later stage.

Problem is, that this detection might be a bit error prone. So for now, lets just skip that.